### PR TITLE
add ci & docker tools

### PR DESCRIPTION
**Pull request by kdada**
Add `Dockerfile`

Add `Jenkinsfile` for ci

### CODE REVIEW

**Comment by ddysher** [comment]
LGTM

@zuomo

**Comment by ddysher** [comment]
@zoumo

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                        make test
+                    ''')
+                }
+            }
+
+            stage("Build image and publish") {
+                sh "docker build -t ${IMAGE_TAG} -f image/Dockerfile ."
+
+                docker.withRegistry("${env.DOCKER_REGISTRY}", "${env.DOCKER_REGISTRY_CREDENTIAL}") {
+                    docker.image("${IMAGE_TAG}").push()
```
Comment (zoumo) [inline Jenkinsfile]: docker.withRegistry(IMAGE_TAG)

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+podTemplate(
+    cloud: 'dev-cluster',
+    namespace: 'kube-system',
+    name: 'helm-registry',
+    label: 'helm-registry',
+    idleMinutes: 1440,
+    containers: [
+        containerTemplate(
+            name: 'jnlp',
+            image: "${env.IMAGE_PREFIX}/circle/jnlp:2.62",
```
Comment (kdada) [inline Jenkinsfile]: It's defined at `Environment variables` in jenkins.

**Comment by zoumo** [comment]
LGTM

**Pull request merged**